pgeocode.py

```diff
@@ -43,41 +35,25 @@ class Nominatim(object):
         into a single entry
     """
     def __init__(self, country='fr', unique=True):
-
-        country = country.upper()
-        if country not in COUNTRIES_VALID:
-            raise ValueError(('country={} is not a known country code. '
-                              'See the README for a list of supported '
-                              'countries')
-                             .format(country))
-        if country == 'AR':
-            warnings.warn("The Argentina data file contains 4-digit postal "
-                          "codes which were replaced with a new system "
-                          "in 1999.")
-        self.country = country
-        self._data_path, self._data = self._get_data(country)
+        country_obj = _Country(country)
+        self.country = country_obj.name
+        self.download_path = country_obj.get_download_path()
```

Comment:
Let's make this private as well,
```suggestion
        self._download_path = country_obj.get_download_path()
```